x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/analyzer/AnalyzerRules.java

```diff
@@ -86,4 +104,139 @@ protected LogicalPlan rule(LogicalPlan plan) {
 
         protected abstract LogicalPlan doRule(LogicalPlan plan);
     }
+
+    public static class ResolveFunctions extends AnalyzerRule<LogicalPlan> {
+
+        private final Configuration configuration;
+        private final FunctionRegistry functionRegistry;
+
+        public ResolveFunctions(Configuration configuration, FunctionRegistry functionRegistry) {
+            this.configuration = configuration;
+            Objects.requireNonNull(functionRegistry);
```

Comment:
I don't think this is necessary. Both these objects are essential and part of the core classes of both EQL and SQL. If until this point they are null, the NPE would have happened already.